### PR TITLE
fix: add placeholder to fix Implicit Else

### PR DESCRIPTION
**Pull request by adrian-burlacu-software**
@bcoe 
While implicit else reporting was fixed in pull #663, it no longer aligns to the instrumented API. Where I previously updated that to provide a placeholder in pull #633, by now the placeholder is gone. 

This became an issue #670 

The simplest was to update the reporting to reflect this change in the instrumenter API. Namely, I added a placeholder into the metadata array when the single if branch with no else condition.

I'd fix the design but that's already taking hours to reason about the various fails I could cause.

### CODE REVIEW

File: packages/istanbul-reports/test/fixtures/github-649.json [hunk 1]
```diff
@@ -100,11 +100,6 @@
             "line": 6,
             "column": 5
           }
-        }, {
```
Comment (bcoe) [inline packages/istanbul-reports/test/fixtures/github-649.json:103]: I'm not understanding why you deleted this from your fixture?

Should we also handle the empty `start`/`end` use case?

File: packages/istanbul-reports/lib/html/annotator.js [hunk 2]
```diff
@@ -142,6 +142,19 @@ function annotateBranches(fileCoverage, structuredText) {
         // only highlight if partial branches are missing or if there is a
         // single uncovered branch.
         if (sumCount > 0 || (sumCount === 0 && branchArray.length === 1)) {
+            // Need to recover the metaArray placeholder item to count an implicit else
+            if (
+                branchMeta[branchName].type === 'if' &&
+                branchArray.length === 2 &&
```
Comment (bcoe) [inline packages/istanbul-reports/lib/html/annotator.js:148]: I think adding a comment for what each of these lines reqpresents would be useful, does `branchArray.length === 2` imply that there's no `else?

**Review by bcoe — COMMENTED** [review]
I left a couple questions, thanks for the patch.

File: packages/istanbul-reports/lib/html/annotator.js (hunk 2 above)
Comment (adrian-burlacu-software) [reply packages/istanbul-reports/lib/html/annotator.js:148]: @bcoe
Will do Monday :)

Yes, the branchArray implies there's only the beginning if and the placeholder in the array.

**Comment by bcoe** [comment]
@adrian-burlacu-software what's the state of this PR? I'm hesitant to land if it requires that we modify an existing regression test. But, since you wrote the regression test, you'd know better than me whether we should land.

**Comment by adrian-burlacu-software** [comment]
> @adrian-burlacu-software what's the state of this PR? I'm hesitant to land if it requires that we modify an existing regression test. But, since you wrote the regression test, you'd know better than me whether we should land.

@bcoe 
Took a look at the files from pull #663  to resolve issue #649. It looks like I left out this placeholder in the test back then, meaning this should get landed.

**Review by bcoe — APPROVED** [review]

**Pull request merged**